File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.Arrays;
import java.util.List;

class ResistorColorDuo {

    private static final List<String> colorList = Arrays.asList("black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white");

    int value(String[] colors) {
        return 10 * colorList.indexOf(colors[0]) + colorList.indexOf(colors[1]);
    }
}

File: javascript/resistor-color-duo/resistor-color-duo.js
const colorCodes = {
  black: 0,
  brown: 1,
  red: 2,
  orange: 3,
  yellow: 4,
  green: 5,
  blue: 6,
  violet: 7,
  grey: 8,
  white: 9,
}

export const decodedValue = colors => 10*colorCodes[colors[0]] + colorCodes[colors[1]]
